modeltranslation/fields.py

```diff
@@ -256,6 +258,17 @@ def __ne__(self, other):
     def __hash__(self):
         return hash((self.creation_counter, self.language))
 
+    def get_default(self):
+        with override(self.language):
+            default = super().get_default()
+            # we must *force evaluation* at this point, otherwise the lazy translatable
+            # string is returned and will be evaluated only later when the main language
+            # is activated again (because this context block has exited).
+            #
+            # force_str passes protected types as-is, which includes None, int, float,
```

Comment:
I wonder how `force_str` handles JSONField, or any other non-standard field.

By the looks of it it will try converting it to string.

Maybe it's better to not use `force_str`, and instead use something like

```
from django.utils.functional import Promise
...

if isinstance(default, Promise):
    default = str(default)
```

Reply:
Looking at https://github.com/django/django/blob/cafe7266ee69f7e017ddbc0d440084ace559b04b/django/utils/encoding.py#L67 with the value of JSONField being a dict for example, this would indeed result in `str(dict)`. The suggested approach is probably more robust. I'll make sure to add a test for this.

I think that it might be better at this point to create a dedicated test model for these default-value cases to not pollute other models and tests, what do you think?

Reply:
For the following supported types of fields, I don't see how a translatable string (via `gettext_lazy` or otherwise) would make sense: IntegerField, BooleanField, FloatField, DecimalField, IPAddressField, GenericIPAddressField, DateField, DateTimeField, TimeField, ForeignKey, ManyToManyField. Sure, you could have different values depending on language, but not something that would benefit from the field language being activated during `get_default`.

That leaves the following fields:

* `CharField`, `TextField`, `FileField` and `ImageField` - straight forward text based fields for which the `force_str` approach works reliably.
* `JSONField`, for which it doesn't work, but we could `isinstance` this and apply the force_str recursively, you only need to support looping over items for arrays, looping over key-values for dicts and handle the primitives, for which you can rely on `force_str` as it handles `None`, numbers and datetime stuff (essentially all the types that django's DjangoJSONEncoder supports).

Reply:
That would complicate things too much. Let's start with what we have already, and fix our approach only if we encounter problems.